### PR TITLE
added github-actions CI

### PR DESCRIPTION
**Pull request by longwuyuan**
This PR attempts to resolve https://github.com/kubernetes-sigs/yaml/issues/48

### CODE REVIEW

**Comment by longwuyuan** [comment]
@dims @palnabarun new clean pr here

**Comment by dims** [comment]
Looks like the GH check passed here : https://github.com/longwuyuan/yaml/actions/runs/512198272

/approve
/lgtm

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+jobs:
+  build:
+    strategy:
+      matrix: 
+        go-versions: [1.12.x, 1.13.x, 1.14.x]
```
Comment (dims) [inline .github/workflows/go.yml:12]: please add newer go 1.15.x in a follow up PR

Comment (longwuyuan) [reply .github/workflows/go.yml:12]: will get on it right away

Comment (longwuyuan) [reply .github/workflows/go.yml:12]: On a different note ;
- Travis CI was explicit about distro and release (ubuntu-xenial) and the github-actions is now using "ubuntu-latest" which is translating to xenial for now. But the GH-Actions is also showing this message ;

```
Ubuntu-latest workflows will use Ubuntu-20.04 soon. For more details, see https://github.com/actions/virtual-environments/issues/1816
```
Does this require any attention or is "ubuntu-latest" ok when GH starts using focal

Comment (longwuyuan) [reply .github/workflows/go.yml:12]: Also, please advise if "only" add 1.15.x or remove 1.12.x as well. thnx

**Pull request merged**